lib/compress/zstd_compress.c

```diff
@@ -2287,6 +2360,10 @@ static size_t ZSTD_compressBlock_internal(ZSTD_CCtx* zc,
             zc->entropyWorkspace, HUF_WORKSPACE_SIZE /* statically allocated in resetCCtx */,
             zc->bmi2);
 
+    if (zc->seqCollector.collectSequences) {
```

Comment:
As mentioned previously,
I believe it's reasonable to consider that,
while collecting sequences, 
users are not expecting compressed data to be actually produced. 
This is, btw, exactly the outcome of `ZSTD_getSequences()`, which throws away the compressed result.

Therefore, don't `ZSTD_compressSequences()`.
Just invoke `ZSTD_copyBlockSequences()` before that point, and bail out with a `0` result.
This will free `ZSTD_getSequences()` from allocating a large memory block.